R/restopt_problem.R

```diff
@@ -0,0 +1,87 @@
+#' @include internal>R
+NULL
+
+# Copyright (c) 2021, Dimitri Justeau-Allaire
+#
+# Institut Agronomique neo-Caledonien (IAC), 98800 Noumea, New Caledonia
+# AMAP, Univ Montpellier, CIRAD, CNRS, INRA, IRD, Montpellier, France
+#
+# This file is part of restoptr
+#
+# restoptr is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# restoptr is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with restoptr  If not, see <https://www.gnu.org/licenses/>.
+
+#' Restoration optimization problem
+#'
+#' Create a new restoration optimization problem using data that describe
+#' the spatial distribution of existing habitat and potential for
+#' restoration.
+#'
+#' @param existing_habitat [terra::rast()] Raster object containing binary
+#' values that indicate if each planning unit contains habitat or not.
+#'
+#' @param restorable_habitat [terra::rast()] Raster object containing
+#' continuous values that indicate the amount of habitat that can
```

Comment:
Just a detail (I can correct it later): restorable values must be integers.

Reply:
Ah - thanks for pointing that out. On that note, maybe it would be worth updating the argument checks to verify that the raster data values are sensible? E.g., no negative values, all values range between sensible values, rasters have at least one non-missing values? We have these checks in the prioritizr package, but they can be slow to run for very large datasets. What do you think?

Reply:
Yes, it would be really nice to have such checks. Maybe we can list these in a specific issue, or keep them in mind to add them once we have the base functionalities of the package working?

Reply:
I've started adding some checks to `restopt_problem` - is there a sensible upper limit for `restorable_habitat`?